pkg/clustermanager/cluster_manager_test.go

```diff
@@ -896,8 +874,8 @@ type clusterChangedTest struct {
 	oldWorkerMachineConfig, newWorkerMachineConfig             *v1alpha1.VSphereMachineConfig
 }
 
-func newClusterChangedTest(t *testing.T) *clusterChangedTest {
-	testSetup := newTest(t)
+func newClusterChangedTest(t *testing.T, opts ...clustermanager.ClusterManagerOpt) *clusterChangedTest {
```

Comment:
nit: Maybe we should rename this method? It was supposed to be only for the `...ClusterChanged` method
but if are finding it useful for other tests, we might want to find a better name
We can do that in a separate PR